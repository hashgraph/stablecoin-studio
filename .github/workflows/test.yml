name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test