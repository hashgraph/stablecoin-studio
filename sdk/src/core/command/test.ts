/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
