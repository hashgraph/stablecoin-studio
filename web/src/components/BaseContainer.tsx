import { Stack, Heading, Box } from '@chakra-ui/react';
import type { ReactNode } from 'react';

interface BaseContainerProps {
	title: string;
	children: ReactNode;
}

const BaseContainer = (props: BaseContainerProps) => {
	const { title, children } = props;
	return (
		<Stack h='full'>
			<Heading
				data-testid='base-container-heading'
				fontSize='28px'
				fontWeight={500}
				lineHeight='35px'
				color='brand.secondary'
				mb='24px'
			>
				{title}
			</Heading>
			<Box data-testid='base-container-children' bg='brand.gray100' h='full' overflowY='hidden'>
				{children}
			</Box>
		</Stack>
	);
};

export default BaseContainer;
