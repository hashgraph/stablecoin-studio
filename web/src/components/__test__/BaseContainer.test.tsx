import { Box } from '@chakra-ui/react';
import { render } from '../../test/index';
import BaseContainer from '../BaseContainer';

const baseContainerProps = {
	title: 'Test',
	children: <Box>Test</Box>,
};

describe(`<${BaseContainer.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<BaseContainer {...baseContainerProps} />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should has title and children visible', () => {
		const component = render(<BaseContainer {...baseContainerProps} />);

		const title = component.getByTestId('base-container-heading');
		const children = component.getByTestId('base-container-children');

		expect(title).toHaveTextContent(baseContainerProps.title);
		expect(children).toBeInTheDocument();
	});
});
