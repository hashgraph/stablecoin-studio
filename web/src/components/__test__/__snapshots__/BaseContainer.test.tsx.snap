// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseContainer /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-16oka5y"
  >
    <h2
      class="chakra-heading css-llrpa2"
      data-testid="base-container-heading"
    >
      Test
    </h2>
    <div
      class="css-1hkemoa"
      data-testid="base-container-children"
    >
      <div
        class="css-0"
      >
        Test
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
